src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`2.cs

```diff
@@ -230,13 +230,24 @@ void AddItemsToBatch(IEnumerable<TItem> items)
                 if (nextBatch.IsEmpty)
                     return default;
 
-                return await _processBatchAsync(nextBatch, batchCancellationToken).ConfigureAwait(false);
-            }
-            catch (OperationCanceledException) when (!_entireQueueCancellationToken.IsCancellationRequested)
-            {
-                // Don't bubble up cancellation to the queue for the nested batch cancellation.  Just because we decided
-                // to cancel this batch isn't something that should stop processing further batches.
-                return default;
+                var batchResultTask = _processBatchAsync(nextBatch, batchCancellationToken).Preserve();
+                await batchResultTask.NoThrowAwaitableInternal(false);
+                if (batchResultTask.IsCompletedSuccessfully)
+                {
+                    return batchResultTask.Result;
+                }
+                else if (batchResultTask.IsCanceled && !_entireQueueCancellationToken.IsCancellationRequested)
+                {
+                    // Don't bubble up cancellation to the queue for the nested batch cancellation.  Just because we decided
+                    // to cancel this batch isn't something that should stop processing further batches.
+                    return default;
+                }
+                else
+                {
+                    // Access the result to force the exception to be thrown.
+                    _ = batchResultTask.Result;
```

Comment:
does this throw cancellation in the event of cancellatino?  or does it throw an aggregate exception?  should this be GetAwaiter().GetResult() instead?

Reply:
➡️ This is a ValueTask, so there is no need to use `GetAwaiter().GetResult()` (exceptions are not wrapped in AggregateException)

Reply:
can you comment this.  needing to know the subtle differences between the task systems is non trivial mental overhead for validating this stuff.